### PR TITLE
feat: add migrations and entities

### LINKED ISSUE #41: [DB Setup] Create the migrations to create the friendship tables
- [x] Define the data model, and document it in the docs of this repository.
  The data model should take into account the use cases for this service, this means storing the following:
  - Friendships
  - Some way to identify a user by their id
  - Friendship requests between two users
  - Friendship history (if a friendship started/ended, store the event)

- [ ] Create the migrations to create the tables

### PR DESCRIPTION
**Pull request by lauti7**
This PR:
- add migrations for DB and run them on service startup
- add entities to query DB
- add helpers to Makefile

Closes #41

### CODE REVIEW

File: src/entities/friendship_history.rs [hunk 1]
```diff
@@ -0,0 +1,49 @@
+//! `SeaORM` Entity. Generated by sea-orm-codegen 0.10.4
+
+use sea_orm::entity::prelude::*;
+
+#[derive(Clone, Debug, PartialEq, DeriveEntityModel, Eq)]
+#[sea_orm(table_name = "friendship_history")]
+pub struct Model {
+    #[sea_orm(primary_key)]
+    pub id: i32,
```
Comment (jmoguilevsky) [inline src/entities/friendship_history.rs:9]: can the id be a random uuid instead of an int?

File: src/entities/friendship_history_events.rs [hunk 2]
```diff
@@ -0,0 +1,25 @@
+//! `SeaORM` Entity. Generated by sea-orm-codegen 0.10.4
+
+use sea_orm::entity::prelude::*;
+
+#[derive(Clone, Debug, PartialEq, DeriveEntityModel, Eq)]
+#[sea_orm(table_name = "friendship_history_events")]
+pub struct Model {
+    #[sea_orm(primary_key)]
+    pub id: i32,
```
Comment (jmoguilevsky) [inline src/entities/friendship_history_events.rs:9]: can the id be a random uuid instead of an int?

File: src/entities/friendships.rs [hunk 3]
```diff
@@ -0,0 +1,27 @@
+//! `SeaORM` Entity. Generated by sea-orm-codegen 0.10.4
+
+use sea_orm::entity::prelude::*;
+
+#[derive(Clone, Debug, PartialEq, DeriveEntityModel, Eq)]
+#[sea_orm(table_name = "friendships")]
+pub struct Model {
+    #[sea_orm(primary_key)]
+    pub id: i32,
```
Comment (jmoguilevsky) [inline src/entities/friendships.rs:9]: can the id be a random uuid instead of an int?

File: src/migrator/m20221125_000001_friendships_table.rs [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+impl MigrationTrait for Migration {
+    // Define how to apply this migration: Create the table.
+    async fn up(&self, manager: &SchemaManager) -> Result<(), DbErr> {
+        manager
+            .create_table(
+                Table::create()
+                    .table(Friendships::Table)
+                    .col(
+                        ColumnDef::new(Friendships::Id)
+                            .integer()
```
Comment (jmoguilevsky) [inline src/migrator/m20221125_000001_friendships_table.rs:21]: same here

File: src/entities/friendship_history.rs [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    pub friendship_id: i32,
+    pub event: i32,
+    pub acting_user: String,
+    pub metadata: Option<Json>,
```
Comment (jmoguilevsky) [inline src/entities/friendship_history.rs:13]: I'd say this should be a JSON, in the worst case it's an option.

**Comment by lauti7** [comment]
Closing this PR. Opening up a new one

**Pull request closed without merging**